web-client/public/js/setup-load-and-import-handlers.js

```diff
@@ -0,0 +1,152 @@
+    $("#errorModal").modal("show");
+};
+
+let reloader = () => { };
+
+export const setupLoadAndImportHandlers = grnState => {
+    const loadGrn = (url, name, formData) => {
+        // The presence of formData is taken to indicate a POST.
+        const fullUrl = [ $("#service-root").val(), url ].join("/");
+        (formData ? $.ajax({
+            url: fullUrl,
+            data: formData,
+            processData: false,
+            contentType: false,
+            type: "POST",
+            crossDomain: true
+        }) : $.getJSON(fullUrl)).done((network, textStatus, jqXhr) => {
+            // Display the network in the console
+            console.log(network);
+            grnState.name = name || jqXhr.getResponseHeader("X-GRNsight-Filename");
+            grnState.network = network;
+
+            reloader = () => loadGrn(url, name, formData);
+
+            updateApp(grnState);
+            // displayStatistics(network);
+        }).error(networkErrorDisplayer);
+    };
+
+    /*
+     * Thanks to http://stackoverflow.com/questions/6974684/how-to-send-formdata-objects-with-ajax-requests-in-jquery
+     * for helping to resolve this.
+     */
+
+    // TODO Some opportunity for unification with loadGrn?
+    const importGrn = (uploadRoute, filename, formData) => {
+        const fullUrl = [ $("#service-root").val(), uploadRoute ].join("/");
+        $.ajax({
+            url: fullUrl,
+            data: formData,
+            processData: false,
+            contentType: false,
+            type: "POST",
+            crossDomain: true
+        }).done(function (network) {
+            grnState.name = filename;
+            grnState.network = network;
+            grnState.annotateLinks();
+
+            reloader = () => importGrn(uploadRoute, filename, formData);
+
+            updateApp(grnState);
+        }).error(networkErrorDisplayer);
+    };
+
+    $("#upload").on("click", () =>
+        // deleted event parameter
+        $("#launchFileOpen").off("click").on("click", () => $("#upload").click())
+    );
+
+    $("#upload-sif").on("change", uploadHandler("upload-sif", importGrn));
+    $("#upload-graphml").on("change", uploadHandler("upload-graphml", importGrn));
+
+    $("#upload").on("change", uploadHandler("upload", loadGrn));
+
+    const loadDemo = url => {
+        loadGrn(url);
+        reloader = () => loadGrn(url);
+
+        $("a.upload > input[type=file]").val("");
+    };
+
+    const initializeDemoFile = (demoId, demoPath, demoName) => {
+        // Deleted parameter `event`
+        $(demoId).on("click", () => loadDemo(demoPath, demoName));
+    };
+
+    DEMO_INFORMATION.forEach(demoInfo => initializeDemoFile.apply(null, demoInfo));
+
+    $("#reload").click(function () {
+        // Deleted `event` parameter but need `function` because of `this`.
+        if (!$(this).parent().hasClass("disabled")) {
+            if ($.isFunction(reloader)) {
+                reloader();
+            }
+        }
+    });
+};
```

Comment:
The rest of this file is almost entirely copied from _upload.js_, the only difference being some code modernization. (`const`s, arrow functions, etc.)